stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/MessageListItemViewHolderFactory.kt

```diff
@@ -13,8 +12,11 @@ import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyMediaAttachm
 import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithFileAttachmentsViewHolder
 import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithMediaAttachmentsViewHolder
 import io.getstream.chat.android.ui.messages.adapter.viewholder.ThreadSeparatorViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.decorator.DecoratorsProvider
 
-public open class MessageListItemViewHolderFactory(private val currentUser: User) {
+public open class MessageListItemViewHolderFactory(
+    private val decoratorsProvider: DecoratorsProvider
+) {
```

Comment:
I think this is a good time to review our customization story for the Message List, I added that as a topic to our Trello board as well earlier today.

So, from here on out, it looks like we'll be allowing clients to:
 
- subclass this factory
- create their own implementation of `DecoratorsProvider` (and use that either in our factory, or their custom one)
- reuse our own existing `DecoratorsProvider` implementations
- create their own `Decorator` implementations

This prompts a bunch of questions:

- Do we need all of these, could we be more restrictive somewhere, expose less API?
- Do we see a chance of clients being able to write meaningful `Decorator`s, with how closely they're tied to the layouts of the ViewHolders?
- If we _do_ expose `Decorator`s, should we make the ones that are used by default `public` as well? They're currently internal.